misk-service/src/main/kotlin/misk/CoordinatedService.kt

```diff
@@ -152,15 +155,11 @@ internal class CoordinatedService(
   }
 
   private fun stopIfReady() {
-    synchronized(this) {
-      if (state() != State.STOPPING) return
-
-      // If any downstream service or its enhancements haven't stopped, don't stop.
-      if (downstreamServices.any { !it.isTerminated() }) return
-
-      if (service.state() != State.RUNNING) return
+    val canStopInner =
+      state() == State.STOPPING && downstreamServices.all { it.isTerminated() }
 
-      // Actually stop.
+    // stopAsync can be called multiple times, with subsequent calls being ignored
+    if (canStopInner) {
```

Comment:
Agreed that internal state check for `service` is not necessary here, since there is no consequence to calling `stopAsync` on a stopped service.